code/modules/agriculture/fishing.dm

```diff
@@ -3,108 +3,123 @@
+					/obj/item/food/fish/swordfish = 25,
+					/obj/item/food/fish/shark = 25,
+					/obj/item/food/fish/great_white_shark = 25)
+	cast_message = "You start harpooning the water!"
+	catch_message = "You harpoon a "
+
+/obj/item/weapon/tool/fishingrod/fly
+	name = "Fly Fishing Rod"
+	desc = "A rod for fly fishing."
+	length = 30
+	fish_given = 1
+	fishables = list(/obj/item/food/fish/trout = 33,
+					/obj/item/food/fish/salmon = 33,
+					/obj/item/food/fish/rainbow_fish = 33)
+	cast_message = "You start fly fishing!"
+	catch_message = "You catch a "
+
+/obj/fishingbouy
 	name = "Fishing Bouy"
 	desc = "Dynamic fish attracting powers!"
 	icon_state = "effect_bouy"
-	length = 30
-	var/fishGiven = 1
-	var/mob/player/fisherman
-
-/obj/effect/fishingbouy/New(var/turf/atloc,var/owner,var/bait)
-	..(atloc)
-	fisherman = owner
-	spawn(1)
-		Beam(owner,time=length,icon_state="f_beam")
-
-/obj/effect/fishingbouy/onDestroy()
-	if(fishGiven > 0)
-		var/A = get_turf(src)
-		var/catchType = pickweight(A:fishables)
-		var/catch = new catchType(get_turf(fisherman))
-		fisherman.playerData.fishing.change(1)
-		messageInfo("You reel in a [catch]",fisherman,src)
-		--fishGiven
-	..()
-
-/turf/floor/outside/liquid
-	var/list/fishables = list(/obj/item/food/fish = 50, /obj/item/food/fish/squid = 50, /obj/item/food/fish/urchin = 25, /obj/item/food/fish/clam = 25, /obj/item/food/fish/shrimp = 75,/obj/item/food/fish/lobster = 15,/obj/item/food/fish/crab = 15, /obj/item/food/fish/sponge = 5)
-
-
-///
+	icon = 'sprite/obj/effects.dmi'
 // FISHIES GLUB GLUB
-///
-
 /obj/item/food/fish
 	name = "Fish"
 	desc = "Floats and glubs."
 	icon = 'sprite/obj/fish.dmi'
 	icon_state = "fish"
 	var/randColour = TRUE
+	var/required_level_fishing = 1
+	var/exp_granted_fishing = 1
 
 /obj/item/food/fish/New()
 	..()
 	if(randColour)
 		color = rgb(rand(20,255),rand(20,255),rand(20,255))
-	var/extraType = pick(/datum/reagent/nutrients,/datum/reagent/rawess,/datum/reagent/paratoxin,/datum/reagent/neurotoxin,/datum/reagent/suffocatetoxin)
-	var/datum/reagent/RE = new extraType
-	reagents.addliquid(RE.id, rand(1,5))
-	for(var/datum/reagent/R in reagents.liquidlist)
-		if(R.id == "ntox" || R.id == "ptox" || R.id == "stox")
-			name = "[pick("Bitter","Sour","Infected","Sick")] [name]"
-		if(R.id == "rawess")
-			name = "Fey [name]"
-
-/obj/item/food/fish/squid
-	name = "Squid"
-	desc = "Bobs and Blibs."
-	icon_state = "squid"
-
-/obj/item/food/fish/urchin
-	name = "Urchin"
-	desc = "Like a sea-mine."
-	icon_state = "urchin"
-
-/obj/item/food/fish/clam
-	name = "Clam"
-	desc = "Portable sea-food in it's own packaging."
-	icon_state = "clam"
-
-/obj/item/food/fish/shrimp
-	name = "Shrimp"
-	desc = "It's other home is a barbeque."
-	icon_state = "shrimp"
-
-/obj/item/food/fish/lobster
-	name = "Lobster"
-	desc = "Pinchy!"
-	icon_state = "lobster"
-
-/obj/item/food/fish/crab
-	name = "Crab"
-	desc = "Pinchy Jr."
-	icon_state = "crab"
-
-/obj/item/food/fish/sponge

```

Comment:
Give me the following for it and I can readd it:

Required Fishing Level
Fishing Experience Granted
Required Cooking Level
Cooking Experience Granted